phoenix-hive/src/main/java/org/apache/phoenix/hive/PhoenixMetaHook.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.hive;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Splitter.MapSplitter;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.metastore.HiveMetaHook;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.phoenix.hive.util.HiveConnectionUtil;
+import org.apache.phoenix.hive.util.HiveConfigurationUtil;
+import org.apache.phoenix.hive.util.HiveTypeUtil;
+import org.apache.phoenix.hive.util.PhoenixUtil;
+
+/**
+* PhoenixMetaHook
+* This class captures all create and delete Hive queries and passes them to phoenix
+*
+* @version 1.0
+* @since   2015-02-08 
+*/
+
+public class PhoenixMetaHook
+  implements HiveMetaHook
+{
+  static Log LOG = LogFactory.getLog(PhoenixMetaHook.class.getName());
+
+  /**
+   *commitCreateTable creates a Phoenix table after the hive table has been created 
+   * incoming hive types.
+   * @param tbl the table properties
+   * 
+   */
+  //Too much logic in this function must revisit and dispatch
+  public void commitCreateTable(Table tbl)
```

Comment:
Actually the idea came from the use of the ES connector; I thought it made life at lot easier when creating the 80% of simple tables. Otherwise you need to go to the 2 systems create each table, nothing hard just adds work and no real value. In a second iteration I wanted to add some more logic here and extract table name from Hive table name, extract PK from the first column or from partition columns.